nav2_core/include/nav2_core/goal_checker.hpp

```diff
@@ -69,7 +69,7 @@ class GoalChecker
   virtual void initialize(
     const rclcpp_lifecycle::LifecycleNode::SharedPtr & nh,
     const std::string & plugin_name) = 0;
-
+  virtual void reset() {}
```

Comment:
I don't think you need to create an empty optional reset override

Reply:
It's been used in dwb_local_planner.cpp as to reset the params in simple_goal_checker.cpp
```c++ line
void
DWBLocalPlanner::setPlan(const nav_msgs::msg::Path & path)
{
  auto path2d = nav_2d_utils::pathToPath2D(path);
  for (TrajectoryCritic::Ptr critic : critics_) {
    critic->reset();
  }

  traj_generator_->reset();
  goal_checker_->reset();

  pub_->publishGlobalPlan(path2d);
  global_plan_ = path2d;
}
```

Reply:
Ah my bad, my first review was a quick once over and I thought this was an override not the declaration.

Disregard.